docs/designers-developers/developers/data/data-core.md

```diff
@@ -617,4 +616,5 @@ _Parameters_
 Action triggered to undo the last edit to
 an entity record, if any.
 
-<!-- END TOKEN(Autogenerated actions) -->
+
+<!-- END TOKEN(Autogenerated actions|../../../../packages/core-data/src/actions.js) -->
```

Comment:
Full-transparency: I'm not entirely sure that the extra newline here would be expected. And, if I recall correctly, these were some of the symptoms we saw in the earlier parallelization efforts that ultimately led to their revert. But when I debugged the code flow of the documentation build script, it was always the case that no two `docgen` processes would run on the same document at the same time, which was the problem we had in the earlier implementation.

Reply:
This is something that I've looked and tested specifically by modifying core-data's selectors and actions, whose changes should be reflected in the package's README and the handbook's data-core doc. It's working as expected: different files are processed asynchronously (good for performance), but different tokens within the same file are processed synchronously (required for docgen). So that's :+1: 

I don't have an answer for the extra line, though. It gives me confidence that it's consistent and doesn't depend on input order: note how the extra line is always added before the end of the second token for all files, while in the `master` implementation that extra line was always added at the end of the first token. In data-core.md the last item is the actions while in the README.md is the selectors. I've also tested for things like input variance (reversed input data and output is the same):

`node ./bin/api-docs/update-api-docs.js packages/core-data/src/selectors.js packages/core-data/src/actions.js`

and 

`node ./bin/api-docs/update-api-docs.js packages/core-data/src/actions.js packages/core-data/src/selectors.js`

produce the same results.